DllImportGenerator/DllImportGenerator/TypePositionInfo.cs

```diff
@@ -16,6 +16,35 @@ namespace Microsoft.Interop
         CharEncoding CharEncoding
     );
 
+    /// <summary>
+    /// Describes how to marshal the contents of a value in comparison to the value itself.
+    /// Only makes sense for array-like types. For example, an "out" array doesn't change the
```

Comment:
>Only makes sense for array-like types.

I'm assuming we have a nice assert or even a Diagnostic warning for the user that one is using `In`/`Out` on a type that we think is silly?

Reply:
Yes we emit a diagnostic warning or error if someone applies `In` or `Out` in a case where the attribute has no effect.